Add to CustomTabsOptions ability to disable opening auth in custom tab

bennycao opened the pull request:
### Changes

Add option in `CustomTabsOptions` to define a list of browser packages where you do not want to launch the authentication url in custom tab. This adds feature https://github.com/auth0/Auth0.Android/issues/805 to solve incompatible custom tab browser.

Please describe both what is changing and why this is important. Include:

- create new method for CustomTabOptions builder `withDisabledCustomTabsPackages` to define list of custom tab packages to disable which assigns to new instance variable `disabledCustomTabsPackages`
- update `CustomTabOptions.toIntent` to get `preferredPackage` which is the preferred browser package calculated by browser picker.
- update `CustomTabOptions.toIntent` to validate if the `preferredPackage` is in the `disabledCustomTabsPackages` list, and if so, just return `Intent` with action of `Intent.ACTION_VIEW` only


### References

solves feature https://github.com/auth0/Auth0.Android/issues/805

### Testing

Please describe how this can be tested by reviewers. Be specific about anything not tested and reasons why. Since this library has unit testing, tests should be added for new functionality and existing tests should complete without errors.

- [x] This change adds unit test coverage

- [ ] This change adds integration test coverage

- [x] This change has been tested on the latest version of the platform/language or why not

### Checklist

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)

- [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)

- [x] All existing and new tests complete without errors

bennycao commented on auth0/src/main/java/com/auth0/android/provider/CustomTabsOptions.java:
```diff
@@ -56,7 +66,12 @@ public static Builder newBuilder() {
 
     @SuppressLint("ResourceType")
-    Intent toIntent(@NonNull Context context, @Nullable CustomTabsSession session) {
+    Intent toIntent(@NonNull Context context, @Nullable CustomTabsSession session, @Nullable String preferredPackage) {
```
arguably could call `getPreferredPackage`

pmathew92 commented:
Hi @bennycao  ,
Thank you for raising this PR. Will review this

pmathew92 commented on auth0/src/main/java/com/auth0/android/provider/CustomTabsOptions.java:
```diff
@@ -44,6 +50,10 @@ boolean hasCompatibleBrowser(@NonNull PackageManager pm) {
         return getPreferredPackage(pm) != null;
     }
 
+    boolean isDisabledCustomTabBrowser(@NonNull String preferredPackage) {
```
Please add comments to this method

bennycao replied:
done

pmathew92 commented on auth0/src/main/java/com/auth0/android/provider/CustomTabsOptions.java:
```diff
@@ -7,18 +7,20 @@
 import com.auth0.android.authentication.AuthenticationException;
 
+import java.util.List;
+
```
Remove unwanted extra lines

bennycao replied:
i'm unsure what you mean by unwanted extra lines, the extra lines are conforming to formatting standards ?

pmathew92 reviewed (commented):
Thank you for raising this PR . Few minor changes  .Please address them and we should be good to merge this

pmathew92 commented on auth0/src/main/java/com/auth0/android/provider/CustomTabsOptions.java:
```diff
@@ -171,14 +193,20 @@ public Builder withBrowserPicker(@NonNull BrowserPicker browserPicker) {
         }
 
+        @NonNull
+        public Builder withDisabledCustomTabsPackages(List<String> disabledCustomTabsPackages) {
```
Please add comments for this builder method

bennycao replied:
done

bennycao commented:
> Hi @bennycao , Thank you for raising this PR. Will review this

@pmathew92 thank you, i've addressed most of your comments. Just one about extra lines which i'm not too sure about it
